### PR TITLE
core/txpool: make transaction validation reusable across packages (pools)

### PR DESCRIPTION
**Pull request by karalabe**
The blobpool coming in a followup PR will be a separare tx pool, but it needs some shared functionality with the main transaction pool. Specifically in the context of this PR, the blobpool will need almost the exact same transaction validations with minor tweaks.

As such, this PR does a reparatory work to make the transaction validation reusable across multiple txpool implementations. The PR also does some minor refactorings to clean up some method names.

A next PR will move tx txpool itself into a legacy "subpool" pool, so I'd like to merge in all functional changes first and have the move into a new package be mostly just code moving and import path reworking (TM).

If you're curious how these are used in the blobpool (to verify the abstraction), this PR contains all the code: https://github.com/ethereum/go-ethereum/pull/26940

### CODE REVIEW

File: core/txpool/txpool.go [hunk 1]
```diff
@@ -598,93 +579,49 @@ func (pool *TxPool) local() map[common.Address]types.Transactions {
...
-	}
-	if tx.GasTipCap().BitLen() > 256 {
-		return core.ErrTipVeryHigh
+	opts := &ValidationOptions{
+		Config: pool.chainconfig,
+		Accept: map[uint8]struct{}{
+			types.LegacyTxType:     {},
+			types.AccessListTxType: {},
+			types.DynamicFeeTxType: {},
+		},
```
Comment (holiman) [inline core/txpool/txpool.go:588]: Some would say that using a map here is overkill, you could use a single byte and use bitflags. E.g. something like this: 

` Accept : (1 << types.LegacyTxType) || ( 1 << types.AccessListTxType)  ||  ( 1 << types.DynamicFeeTxType) `

File: core/txpool/validation.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+			return fmt.Errorf("invalid number of %d blobs compared to %d blob hashes", len(blobs), len(hashes))
+		}
+		if len(commits) != len(hashes) {
+			return fmt.Errorf("invalid number of %d blob commitments compared to %d blob hashes", len(commits), len(hashes))
+		}
+		if len(proofs) != len(hashes) {
+			return fmt.Errorf("invalid number of %d blob proofs compared to %d blob hashes", len(proofs), len(hashes))
+		}
+		// Blob quantities match up, validate that the provers match with the
+		// transaction hash before getting to the creyptography
```
Comment (holiman) [inline core/txpool/validation.go:132]:
```suggestion
		// transaction hash before getting to the cryptography
```

**Review by holiman — APPROVED** [review]
LGTM

File: core/txpool/txpool.go (hunk 1 above)
Comment (karalabe) [reply core/txpool/txpool.go:588]: Fair enough, I'll update it

**Comment by karalabe** [comment]
@holiman Fixed, PTAL

**Pull request merged**